include/podio/ASCIIWriter.h

```diff
@@ -1,81 +1,85 @@
-#ifndef ASCIIWRITER_H
-#define ASCIIWRITER_H
+#ifndef PODIO_ASCIIWRITER_H
+#define PODIO_ASCIIWRITER_H
```

Comment:
As mentioned by Frank:  Spelling `ASCIIWRITER` like the classname here (`PODIO_ASCIIWriter_H`) would make search and replace operations a bit more convenient

Reply:
Unfortunately the [llvm-header-guard check](https://clang.llvm.org/extra/clang-tidy/checks/llvm-header-guard.html) is not configurable and simply enforces the [LLVM header guard style](https://llvm.org/docs/CodingStandards.html#header-guard), and there doesn't seem to be anything similar with more freedom. So our choices are:
- Drop the check, resulting in no check for the header guards.
- Keep the check, and have more work in search and replace.

I, personally, would go for the first option here, since this is one of the checks that can be automatically fixed by `clang-tidy` (using the `--fix` option when calling `clang-tidy`). However, if more people prefer the second option removing this check is trivial.

Reply:
I agree with Thomas. This automatic check and fix means one doesn't have to replace them themselves. And automatic unique header guards prevent real issues.

Also: emacs by default does a case aware replacement, so it would replace an all upper case word by an all upper case word. (CamelCasing is more an issue though)
```
ASCIIWRITER
ASCIIWriter

replace asciiwriter -> UTF8Reader gives

UTF8READER
UTF8Reader
```

Reply:
Real world issue: https://github.com/AIDASoft/DD4hep/pull/672

Reply:
Fine with me. Just had gotten to use the same camel case for the header guards as in the class name for search and replace. We should than activate the `--fix` for the header guards...

Reply:
The `--fix` option is "global" for `clang-tidy`, so it will try to fix all the things that it can fix. In general, during CI we are running all checkers with the automatic fixes turned off. But the main point is that it is possible to run `clang-tidy` and `clang-format` locally in a way that fixes are applied automatically.